src/ng2-smart-table/components/cell/cell-edit-mode/custom-edit.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, ComponentFactoryResolver, ViewChild, ViewContainerRef, SimpleChanges } from '@angular/core';
+
+import { EditCellDefault } from './edit-cell-default';
+import { Cell } from '../../../lib/data-set/cell';
+
+@Component({
+  selector: 'table-cell-custom-editor',
+  template: `
+    <template #dynamicTarget></template>
+  `,
+})
+export class CustomEditComponent extends EditCellDefault {
+
+  customComponent: any;
+  @ViewChild('dynamicTarget', {read: ViewContainerRef}) dynamicTarget: any;
+
+  constructor(private resolver: ComponentFactoryResolver) {
+    super();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
Ok, so we should consider doing the same on all the other components eg. https://github.com/akveo/ng2-smart-table/blob/master/demo/src/app/pages/demo/demo.component.ts ?